spec/models/principal_spec.rb

```diff
@@ -202,4 +202,325 @@
       expect(Firm.where(id: firm.id)).to be_empty
     end
   end
+
+  describe '#next_onboarding_action' do
+    context 'when principal has no parent firm' do
+      before :each do
+        principal.firm.destroy
+        principal.reload
+      end
+
+      it 'returns :complete_a_firm' do
+        expect(principal.next_onboarding_action).to eql(:complete_a_firm)
+      end
+    end
+
+    context 'when principal only has a parent firm' do
+      before :each do
+        expect(principal.firm).not_to be_nil
+        expect(Firm.where(fca_number: principal.fca_number).count).to eql(1)
+      end
+
+      context 'and the parent firm has no email address' do
+        before :each do
+          expect(principal.firm.email_address).to be_blank
+        end
+
+        it 'returns :complete_a_firm' do
+          expect(principal.next_onboarding_action).to eql(:complete_a_firm)
+        end
+      end
+
+      context 'and the parent firm has an email address' do
+        before :each do
+          principal.firm.update_column(:email_address, 'acme@example.com')
+          principal.reload
+        end
+
+        context 'and the firm primarily gives remote advice' do
+          before :each do
+            principal.firm.in_person_advice_methods.destroy_all
+            principal.firm.other_advice_methods << create(:other_advice_method)
+            principal.reload
+          end
+
+          context 'and the firm has no advisers' do
+            before :each do
+              expect(principal.firm.advisers.any?).to eql(false)
+            end
+
+            it 'returns :onboarded' do
+              expect(principal.next_onboarding_action).to eql(:onboarded)
+            end
+          end
+
+          context 'and the firm has advisers' do
+            before :each do
+              create(:adviser, firm: principal.firm)
+              principal.reload
+              expect(principal.firm.advisers.any?).to eql(true)
+            end
+
+            it 'returns :onboarded' do
+              expect(principal.next_onboarding_action).to eql(:onboarded)
+            end
+          end
+        end
+
+        context 'and the firm primarily gives in-person advice' do
+          before :each do
+            principal.firm.in_person_advice_methods << create(:in_person_advice_method)
+            principal.reload
+          end
+
+          context 'and has at least one in-person adviser' do
```

Comment:
There is no such thing as an "in-person adviser", can we remove the "in-person" prefix or clarify what we mean here?